alica_standard_library/src/Conditions.cpp

```diff
@@ -93,4 +93,85 @@ bool IsChildSuccess(const alica::Blackboard* input, const alica::RunningPlan* rp
     }
     return false;
 }
+
+template <typename T>
+bool IsEqual(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    try {
+        alica::LockedBlackboardRO bb(*input);
+        const T left = bb.get<T>("left");
+        const T right = bb.get<T>("right");
+        return left == right;
+    } catch (...) {
+        alica::Logging::logError(__func__) << "Error while evaluating IsEqual condition. Please check the input types";
```

Comment:
this should never be hit because alica verifies the types during key mapping itself. If it is then it might be a bug.